### PR TITLE
#load a .cs file without preprocessor or other hacking

### PR DESCRIPTION
**Pull request by StingyJack**
This lets a  .cs file be loaded and used as script.  For me this solves some of the preprocessor directive hacking I was doing to reuse existing classes or get intellisense. (ideas gleaned from issue  #36 )

There is a flowchart and some other docs that describe it more completely but the gist is that a derived SourceFileResolver is passed into the ScriptOptions, and it intercepts the calls to .cs files and uses SyntaxTree et.al. to provide a "namespace free" compiled version of the source that can be lifted into the script scope.

There are a few orphaned bits in the CsRewriter class that I left in because I think adding debugging support will require them.

I tried to not pollute this with the OutputBehavior PR. Please review and suggest any edits, corrections, or additions.

Example of the script with #load for .cs
```
#r ".\..\..\..\packages\NUnit.3.4.0\lib\net45\nunit.framework.dll"
#load "..\TestCs\ReferencedClass.cs"
#load "ReferencedScript.csx"

//Write using supplied ScriptContext
Output.WriteLine("namespace TestNamespace{class TestClass{public void TestMethod(){}}}");

//with the referenced assembly
var bag = new NUnit.Framework.Internal.PropertyBag();
Output.WriteLine($"// {bag}");

//With referenced script
Output.WriteLine($"// we have multiplied {Go4th(2, 3)}");

//Create instance from intercepted class
var rc1 = new ReferencedClass(Context);
Output.WriteLine($"// Emitting prop with backing field {rc1.PropertyWithBackingField}");
rc1.Owl($"// using the referenced class to output")
```

### CODE REVIEW

File: src/Scripty.CustomTool/BaseCodeGenerator.cs [hunk 1]
```diff
@@ -1,4 +1,6 @@
-﻿
+﻿//Dave Glick - is this just template copyright stuff that was brought in or 
+// is this something else?
+
```
Comment (daveaglick) [inline src/Scripty.CustomTool/BaseCodeGenerator.cs:3]: Yeah, the two files with this copyright header were copied wholesale from a Microsoft example.

**Comment by daveaglick** [comment]
Thanks. This pull request has been integrated manually.

**Comment by daveaglick** [comment]
Went ahead and merged this one, everything looks good. Unfortunately I'm having tons of other problems with the new Roslyn assemblies, so a release will have to wait a bit, but I'll get it out as soon as I can get everything working.

File: src/Scripty.CustomTool/BaseCodeGenerator.cs (hunk 1 above)
Comment (StingyJack) [reply src/Scripty.CustomTool/BaseCodeGenerator.cs:3]: Some of these classes are included in the newer VS shell references. You may be able to remove them if you up the shell ref's.

Comment (daveaglick) [reply src/Scripty.CustomTool/BaseCodeGenerator.cs:3]: Good call - planning on continuing work on updating everything tomorrow so I'll take a look and see if they can be stripped.

**Pull request merged**